diff --git a/mlx/c/fft.cpp b/mlx/c/fft.cpp
index cd63ccb..0618068 100644
--- a/mlx/c/fft.cpp
+++ b/mlx/c/fft.cpp
@@ -17,7 +17,7 @@ extern "C" int mlx_fft_fft(
   try {
     mlx_array_set_(
         *res,
-        mlx::core::fft::fft(mlx_array_get_(a), n, axis, mlx_stream_get_(s)));
+        mlx::core::fft::fft(mlx_array_get_(a), n, axis, mlx::core::fft::FFTNorm::Backward, mlx_stream_get_(s)));
   } catch (std::exception& e) {
     mlx_error(e.what());
     return 1;
@@ -39,7 +39,7 @@ extern "C" int mlx_fft_fft2(
             mlx_array_get_(a),
             mlx::core::Shape(n, n + n_num),
             std::vector<int>(axes, axes + axes_num),
-            mlx_stream_get_(s)));
+            mlx::core::fft::FFTNorm::Backward, mlx_stream_get_(s)));
   } catch (std::exception& e) {
     mlx_error(e.what());
     return 1;
@@ -61,7 +61,7 @@ extern "C" int mlx_fft_fftn(
             mlx_array_get_(a),
             mlx::core::Shape(n, n + n_num),
             std::vector<int>(axes, axes + axes_num),
-            mlx_stream_get_(s)));
+            mlx::core::fft::FFTNorm::Backward, mlx_stream_get_(s)));
   } catch (std::exception& e) {
     mlx_error(e.what());
     return 1;
@@ -96,7 +96,7 @@ extern "C" int mlx_fft_ifft(
   try {
     mlx_array_set_(
         *res,
-        mlx::core::fft::ifft(mlx_array_get_(a), n, axis, mlx_stream_get_(s)));
+        mlx::core::fft::ifft(mlx_array_get_(a), n, axis, mlx::core::fft::FFTNorm::Backward, mlx_stream_get_(s)));
   } catch (std::exception& e) {
     mlx_error(e.what());
     return 1;
@@ -118,7 +118,7 @@ extern "C" int mlx_fft_ifft2(
             mlx_array_get_(a),
             mlx::core::Shape(n, n + n_num),
             std::vector<int>(axes, axes + axes_num),
-            mlx_stream_get_(s)));
+            mlx::core::fft::FFTNorm::Backward, mlx_stream_get_(s)));
   } catch (std::exception& e) {
     mlx_error(e.what());
     return 1;
@@ -140,7 +140,7 @@ extern "C" int mlx_fft_ifftn(
             mlx_array_get_(a),
             mlx::core::Shape(n, n + n_num),
             std::vector<int>(axes, axes + axes_num),
-            mlx_stream_get_(s)));
+            mlx::core::fft::FFTNorm::Backward, mlx_stream_get_(s)));
   } catch (std::exception& e) {
     mlx_error(e.what());
     return 1;
@@ -175,7 +175,7 @@ extern "C" int mlx_fft_irfft(
   try {
     mlx_array_set_(
         *res,
-        mlx::core::fft::irfft(mlx_array_get_(a), n, axis, mlx_stream_get_(s)));
+        mlx::core::fft::irfft(mlx_array_get_(a), n, axis, mlx::core::fft::FFTNorm::Backward, mlx_stream_get_(s)));
   } catch (std::exception& e) {
     mlx_error(e.what());
     return 1;
@@ -197,7 +197,7 @@ extern "C" int mlx_fft_irfft2(
             mlx_array_get_(a),
             mlx::core::Shape(n, n + n_num),
             std::vector<int>(axes, axes + axes_num),
-            mlx_stream_get_(s)));
+            mlx::core::fft::FFTNorm::Backward, mlx_stream_get_(s)));
   } catch (std::exception& e) {
     mlx_error(e.what());
     return 1;
@@ -219,7 +219,7 @@ extern "C" int mlx_fft_irfftn(
             mlx_array_get_(a),
             mlx::core::Shape(n, n + n_num),
             std::vector<int>(axes, axes + axes_num),
-            mlx_stream_get_(s)));
+            mlx::core::fft::FFTNorm::Backward, mlx_stream_get_(s)));
   } catch (std::exception& e) {
     mlx_error(e.what());
     return 1;
@@ -235,7 +235,7 @@ extern "C" int mlx_fft_rfft(
   try {
     mlx_array_set_(
         *res,
-        mlx::core::fft::rfft(mlx_array_get_(a), n, axis, mlx_stream_get_(s)));
+        mlx::core::fft::rfft(mlx_array_get_(a), n, axis, mlx::core::fft::FFTNorm::Backward, mlx_stream_get_(s)));
   } catch (std::exception& e) {
     mlx_error(e.what());
     return 1;
@@ -257,7 +257,7 @@ extern "C" int mlx_fft_rfft2(
             mlx_array_get_(a),
             mlx::core::Shape(n, n + n_num),
             std::vector<int>(axes, axes + axes_num),
-            mlx_stream_get_(s)));
+            mlx::core::fft::FFTNorm::Backward, mlx_stream_get_(s)));
   } catch (std::exception& e) {
     mlx_error(e.what());
     return 1;
@@ -279,7 +279,7 @@ extern "C" int mlx_fft_rfftn(
             mlx_array_get_(a),
             mlx::core::Shape(n, n + n_num),
             std::vector<int>(axes, axes + axes_num),
-            mlx_stream_get_(s)));
+            mlx::core::fft::FFTNorm::Backward, mlx_stream_get_(s)));
   } catch (std::exception& e) {
     mlx_error(e.what());
     return 1;
